v2/call_option.go

```diff
@@ -63,6 +63,52 @@ func WithRetry(fn func() Retryer) CallOption {
 	return retryerOption(fn)
 }
 
+// OnErrors returns a Retryer that retries if and only if the previous attempt
+// returns an error that matches one of the errors listed in errs when compared
+// with the given compare function. Supplying errors.Is (https://pkg.go.dev/errors#Is)
+// for compare is valid.
+//
+// Pause times between retries are specified by bo. bo is only used for its
+// parameters; each Retryer has its own copy.
+func OnErrors(bo Backoff, compare func(err, target error) bool, errs ...error) Retryer {
+	return &errRetryer{
+		backoff: bo,
+		shouldRetry: func(err error) bool {
+			for _, e := range errs {
+				if compare(err, e) {
+					return true
+				}
+			}
+			return false
+		},
+	}
+}
+
+// OnError returns a Retryer that retries if and only if the previous attempt
+// returns an error that satisfies shouldRetry.
+//
+// Pause times between retries are specified by bo. bo is only used for its
+// parameters; each Retryer has its own copy.
+func OnError(bo Backoff, shouldRetry func(err error) bool) Retryer {
```

Comment:
This seems great. I'm also trying to incorporate idempotency into my strategy for storage-- maybe we should talk about how these pieces will fit together.

Reply:
Also, I'm not sure if `OnError` is the right name here given that this can cover a whole range of errors? `OnPredicate` maybe?

Reply:
>  not sure if OnError is the right name here given that this can cover a whole range of errors

Yeah I agree, doesn't really feel completely right. @tbpg @codyoss WDYT?

Reply:
`OnPredicate` is fine with me. I will switch it.

Reply:
Done.

Reply:
What about `gax.ErrorRetryer`? `gax.OnPredicate` doesn't say what it is.

Reply:
So this is the name of the call option for the Retryer, and I wonder if we should follow `OnCodes` here (not that 1 instance is a standard, but...) with the `On` prefix. That's why I started with `OnError`.

Reply:
Switched to `OnErrorFunc`